metabase/metabase
# Fix DX on JavaScripts files and Cypress tests

> **Note**
> This is very related to https://github.com/metabase/metabase/pull/28749, I hope we can merge that PR before this one, so I could refactor this PR to only contain a single additional `tsconfig.json` for Cypress directory.

### Motivation
As a VS Code users, I had a lot of painful experience dealing with some JavaScript files and Cypress test files until now. Code navigation is a hit or miss in some JS files it could be hard to find where the functions/components are defined or even being used, and auto import is also not working sometimes.

Also, in Cypress tests, code navigation or autoimport have never worked. As we have a ton of commands living under `e2e/support` it would be ideal to be able to jump right into where those functions are defined.

And this PR attempts to fix that.

### Description
Load both JS and TS files into TypeScript, so code navigation works on JS files as well, and add a new `tsconfig.json` under `e2e` which contains different types for Cypress.

### How to verify
First, test with VS Code, or similar IDE. I've seen Webstorm already works without changing anything, and @nemanjaglumac  told me that IntelliJ also works fine.

1. Try to autoimport or jump to definition in Cypress tests.
2. Try to autoimport or jump to definition in JavaScript files.
3. Try to shift+F12 to see the usage of exported variables in JavaScript files.
4. `yarn type-check` should still work
5. CI checks are green

### Demo
To be added later.
